google-cloud-compute-v1/samples/quickstart.rb

```diff
@@ -0,0 +1,192 @@
+# frozen_string_literal: true
+
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# [START compute_instances_create]
+# [START compute_instances_delete]
+# [START compute_instances_list]
+# [START compute_instances_list_all]
+# [START compute_instances_operation_check]
+
+require "google/cloud/compute/v1"
+
+# [END compute_instances_operation_check]
+# [END compute_instances_list_all]
+# [END compute_instances_list]
+# [END compute_instances_delete]
+# [END compute_instances_create]
+
+# [START compute_instances_create]
+
+# Sends an instance creation request to the Compute Engine API and waits for it to complete.
+#
+# @param [String] project project ID or project number of the Cloud project you want to use.
+# @param [String] zone name of the zone you want to use. For example: “us-west3-b”
+# @param [String] instance_name name of the new virtual machine.
+# @param [String] machine_type machine type of the VM being created. This value uses the
+#         following format: "zones/{zone}/machineTypes/{type_name}".
+#         For example: "zones/europe-west3-c/machineTypes/f1-micro"
+# @param [String] source_image path to the operating system image to mount on your boot
+#         disk. This can be one of the public images
+#         (like "projects/debian-cloud/global/images/family/debian-10")
+#         or a private image you have access to.
+# @param [String] network_name name of the network you want the new instance to use.
+#         For example: "global/networks/default" represents the `default`
+#         network interface, which is created automatically for each project.
+# @return Instance object.
+def create_instance project:, zone:, instance_name:,
+                    machine_type: "n2-standard-2",
+                    source_image: "projects/debian-cloud/global/images/family/debian-10",
+                    network_name: "global/networks/default"
+  client = ::Google::Cloud::Compute::V1::Instances::Rest::Client.new
+
+  disk = ::Google::Cloud::Compute::V1::AttachedDisk.new
```

Comment:
I know the Python samples use these proto classes and construct proto objects when creating requests. I think the main reason they do so is for IDE autocompletion. However, I think we should use nested hashes to represent requests in the Ruby samples. It's much easier to read, more consistent with our other samples, and feels more Ruby-like.